etc/systemd/system-generators/zfs-mount-generator.in

```diff
@@ -230,15 +313,87 @@ ExecStop=@sbindir@/zfs unload-key '${dataset}'"   > "${dest_norm}/${keyloadunit}
       "${dataset}" >/dev/kmsg
   fi
 
-  # If the mountpoint has already been created, give it precedence.
+  if [ -n "${p_systemd_wantedby}" ] && \
+      [ "${p_systemd_wantedby}" != "-" ] ; then
+    p_systemd_noauto="on"
```

Comment:
We shouldn't actually do this (even though it makes it much simpler). Setting it `noauto` will cause the file to take lower precedence in your handling of existing files---I guess you should make another flag here.

Reply:
It actually doesn't, I've spent some time thinking about this very thing.
Sorting and [not] overwriting are done on `p_canmount`. The only line in the file overwriting section where we check against `p_systemd_noauto` instead of canmount is whether to log, line 380.